packages/xrpl/src/sugar/autofill.ts

```diff
@@ -280,22 +279,21 @@ export async function calculateFeePerTransactionType(
     tx.TransactionType === 'AccountDelete' ||
     tx.TransactionType === 'AMMCreate'
   ) {
-    baseFee = await fetchAccountDeleteFee(client)
+    baseFee = await fetchOwnerReserveFee(client)
   }
 
   /*
    * Multi-signed Transaction
-   * 10 drops × (1 + Number of Signatures Provided)
+   * BaseFee × (1 + Number of Signatures Provided)
    */
   if (signersCount > 0) {
     baseFee = BigNumber.sum(baseFee, scaleValue(netFeeDrops, 1 + signersCount))
   }
 
   const maxFeeDrops = xrpToDrops(client.maxFeeXRP)
-  const totalFee =
-    tx.TransactionType === 'AccountDelete'
-      ? baseFee
-      : BigNumber.min(baseFee, maxFeeDrops)
+  const totalFee = ['AccountDelete', 'AMMCreate'].includes(tx.TransactionType)
```

Comment:
What is the need for this comparison with `maxFeeDrops` ( == `2,000,000` drops) ?

the `baseFee` variable computes the final fees, accounting for the cases where multiple-signers are provided. 

Why are performing this arbitrary comparison with `2 XRP` ? Moreover, we are not using any `cushion` value in the `getFeeXrp()` function call, hence there is no possibility of the library over-estimating the fees.

Reply:
This part confuses me a bit as well since baseFee computed the final fee with multi-signers accounted so not sure why we need this comparison.
@khancode could you explain why since you wrote this code a while ago?
Btw, cushion is defaulted to 1.2 if not set.